source/common/common/interval_value.h

```diff
@@ -22,6 +22,13 @@ template <typename T, typename Interval> class ClosedIntervalValue {
 
   T value() const { return value_; }
 
+  bool operator==(ClosedIntervalValue<T, Interval> other) const { return value_ == other.value(); }
```

Comment:
Any concerns about  rounding errors in float comparison?

Reply:
No, the endpoints (0 and 1) are represented exactly by floats. Other code shouldn't use this. I guess that might be an argument for not having == or != ?

Reply:
Given that this is just a refactor, I think this isn't any worse than before.

Reply:
I'm a little paranoid, because later in the PR i see this pattern:
```
float x = 1.0;
float y = 0.0;
float z = 1 - y;
flow w = 1 - z;
```
Are we then guaranteed that  x==z and w==y?

In other words, are the exact comparisons guaranteed after there's been arithmetic in the flow? Can you check if you haven't already?

Reply:
One thing you could consider is offering an isMax() or isMin() method which could abstract a more robust way of doing compares.

Reply:
No, there's no guarantee that 1.0f == (1.0f - 0.0f). I've gone ahead and removed the equality comparison operators in favor of isMin and isMax methods.